api/src/main/java/org/openmrs/module/addresshierarchy/config/AddressConfiguration.java

```diff
@@ -0,0 +1,157 @@
+package org.openmrs.module.addresshierarchy.config;
+
+import java.lang.reflect.Constructor;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.beanutils.MethodUtils;
+import org.openmrs.api.APIException;
+import org.openmrs.api.context.Context;
+import org.openmrs.util.OpenmrsUtil;
+
+/**
+ * Simple component for representing the configuration options when starting up the address hierarchy module.
+ */
+public class AddressConfiguration {
+
+    // Properties - Those are the fields that must be in the config. XML.
+
+	private boolean wipe = false;
+    private List<AddressComponent> addressComponents;
+    private List<String> lineByLineFormat;
+    private AddressHierarchyFile addressHierarchyFile;
+
+    // Constructor
+
+    public AddressConfiguration() {}
+
+    // Accessors
+    
+    public boolean mustWipe() {
+    	return wipe;
+    }
+
+    public List<AddressComponent> getAddressComponents() {
+        if (addressComponents == null) {
+            addressComponents = new ArrayList<AddressComponent>();
+        }
+        return addressComponents;
+    }
+
+    public void addAddressComponent(AddressComponent component) {
+        getAddressComponents().add(component);
+    }
+
+    public void setAddressComponents(List<AddressComponent> addressComponents) {
+        this.addressComponents = addressComponents;
+    }
+
+    public List<String> getLineByLineFormat() {
+        if (lineByLineFormat == null) {
+            lineByLineFormat = new ArrayList<String>();
+        }
+        return lineByLineFormat;
+    }
+
+    public void addLineByLineFormat(String line) {
+        getLineByLineFormat().add(line);
+    }
+
+    public void setLineByLineFormat(List<String> lineByLineFormat) {
+        this.lineByLineFormat = lineByLineFormat;
+    }
+
+    public AddressHierarchyFile getAddressHierarchyFile() {
+        return addressHierarchyFile;
+    }
+
+    public void setAddressHierarchyFile(AddressHierarchyFile addressHierarchyFile) {
+        this.addressHierarchyFile = addressHierarchyFile;
+    }
+
+    // Instance methods
+
+    /**
+     * @return a new AddressTemplate instance for the given configuration
+     */
+    public Object getAddressTemplate() {
+        Object addressTemplate = null;
```

Comment:
This is fine I guess, though since the only purpose of this is to generate serialized XML, I think I'd be more inclined to have a new "AddressTemplate" class that is a part of the module, and to simply have a custom serializer instance that is capable of serializing this into the proper format expected by any version of OpenMRS.